src/main/java/org/folio/rmapi/RMAPIService.java

```diff
@@ -180,6 +184,36 @@ public RMAPIService(String customerId, String apiKey, String baseURI, Vertx vert
     return this.getRequest(constructURL(TITLES_PATH + "?" + path), Titles.class);
   }
 
+  public CompletableFuture<Packages> retrievePackages(
+    String filterSelected, String filterType, Integer id, String q, int page, int count,
+    Sort sort) {
+    String path = new PackagesFilterableUrlBuilder()
+      .filterSelected(filterSelected)
+      .filterType(filterType)
+      .q(q)
+      .page(page)
+      .count(count)
+      .sort(sort)
+      .build();
+
+    String packagesPath = id == null ? PACKAGES_PATH + "?" : VENDORS_PATH+ '/' + id + '/' + PACKAGES_PATH + "?";
+
+    return this.getRequest(constructURL(packagesPath + path), Packages.class);
+  }
+
+  public CompletableFuture<Vendors> getVendors(boolean filterCustom){
+    CompletableFuture<Vendors> vendorsList = CompletableFuture.completedFuture(new Vendors());
+    if (Boolean.TRUE.equals(filterCustom)) {
```

Comment:
This can be simplified to 
if (filterCustom)
similary in PackageParametersValidator condition can be simplified